pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeTableDataManager.java

```diff
@@ -452,6 +460,12 @@ public void addSegment(String segmentName, IndexLoadingConfig indexLoadingConfig
     _serverMetrics.addValueToTableGauge(_tableNameWithType, ServerGauge.SEGMENT_COUNT, 1L);
   }
 
+  private boolean isReadyToPreload() {
```

Comment:
Rename this to `isUpsertPreloadEnabled()`. We don't need to check both upsert and this, checking only this should be enough.